### PR TITLE
multi: extend account expiry after using it

### PR DESCRIPTION
**Pull request by positiveblue**
Follows the same format than in `RenewAccount`: one of `absolute_expiry` or `relative_expiry`.

`--expiry_height` or `--expiry_blocks` in `pool account`

### CODE REVIEW

File: cmd/pool/account.go [hunk 1]
```diff
@@ -417,6 +418,17 @@ var depositAccountCommand = cli.Command{
 			Usage: "the fee rate expressed in sat/vbyte that " +
 				"should be used for the deposit",
 		},
+		cli.Uint64Flag{
+			Name: accountExpiryAbsolute,
+			Usage: "the new block height at which this account " +
```
Comment (guggero) [inline cmd/pool/account.go:423]: nit: remove first occurrence of "at"?

Comment (guggero) [reply cmd/pool/account.go:423]: Could also define the whole flag as a variable and then use it twice?

File: rpcserver.go [hunk 2]
```diff
@@ -746,11 +746,22 @@ func (s *rpcServer) DepositAccount(ctx context.Context,
 			"minimum is %d sat/kw", feeRate, chainfee.FeePerKwFloor)
 	}
 
+	bestHeight := atomic.LoadUint32(&s.bestHeight)
+
+	// If provided, determine new expiration value.
+	var expiryHeight uint32
+	switch {
+	case req.GetAbsoluteExpiry() != 0:
```
Comment (guggero) [inline rpcserver.go:754]: We also need to add the check for both being set, in case someone uses the API directly. Same for the withdrawal.

File: account/manager_test.go [hunk 3]
```diff
@@ -869,7 +869,7 @@ func TestAccountWithdrawal(t *testing.T) {
 	dustOutput := &wire.TxOut{Value: 0, PkScript: p2wsh}
 	_, _, err := h.manager.WithdrawAccount(
 		context.Background(), account.TraderKey.PubKey,
-		[]*wire.TxOut{dustOutput}, feeRate, bestHeight,
+		[]*wire.TxOut{dustOutput}, feeRate, bestHeight, 0,
```
Comment (guggero) [inline account/manager_test.go:872]: Should we also add a test case for a non-zero expiration that checks the account modifications are added correctly?

**Review by guggero — COMMENTED** [review]
Nice work, was able to test this locally and it all worked great!
Only a few minor comments from my end. The only one blocking IMO is the unit test case.
Other than that LGTM :tada:

File: account/manager_test.go (hunk 3 above)
Comment (positiveblue) [reply account/manager_test.go:872]: totally agree. The tests are currently pretty coupled to mock/hard coded values so let's add it after #313 has been merged and we can manage the expected behavior better.

File: cmd/pool/account.go (hunk 1 above)
Comment (positiveblue) [reply cmd/pool/account.go]: done (it was shared with other cmds too)

File: cmd/pool/account.go [hunk 4]
```diff
@@ -417,6 +418,13 @@ var depositAccountCommand = cli.Command{
 			Usage: "the fee rate expressed in sat/vbyte that " +
 				"should be used for the deposit",
 		},
+		expiryAbsoluteFlag,
+		cli.Uint64Flag{
```
Comment (guggero) [inline cmd/pool/account.go:422]: nit: extract this flag into a variable as well?

**Review by guggero — APPROVED** [review]
Nice work, LGTM :tada:

File: cmd/pool/account.go (hunk 4 above)
Comment (positiveblue) [reply cmd/pool/account.go:422]: I thought about it but one out of the three usages has a default value set and the other two don't... we can have a separated PR consolidating the flags later.

File: rpcserver.go [hunk 5]
```diff
@@ -746,11 +746,25 @@ func (s *rpcServer) DepositAccount(ctx context.Context,
 			"minimum is %d sat/kw", feeRate, chainfee.FeePerKwFloor)
 	}
 
+	bestHeight := atomic.LoadUint32(&s.bestHeight)
+
+	// If provided, determine new expiration value.
+	var expiryHeight uint32
+	switch {
+	case req.GetAbsoluteExpiry() != 0 && req.GetRelativeExpiry() != 0:
```
Comment (bhandras) [inline rpcserver.go:754]: nit: is this case even possible? (given it's a `oneof` in the proto).

**Review by bhandras — APPROVED** [review]
LGTM :+1:

File: rpcserver.go (hunk 5 above)
Comment (guggero) [reply rpcserver.go:754]: Ah, you're right, it can't really. Forgot it was a `oneof` when I asked for this case to be added. But doesn't hurt either, so we can just leave it IMO.
